osu.Game.Rulesets.Osu/Objects/Drawables/Connections/FollowPointConnection.cs

```diff
@@ -131,10 +139,13 @@ private void refresh()
                     fp.ScaleTo(osuEnd.Scale, osuEnd.TimeFadeIn, Easing.Out);
                     fp.MoveTo(pointEndPosition, osuEnd.TimeFadeIn, Easing.Out);
                     fp.Delay(fadeOutTime - fadeInTime).FadeOut(osuEnd.TimeFadeIn);
-                }
 
-                fp.Expire(true);
+                    finalTransformEndTime = fadeOutTime + osuEnd.TimeFadeIn;
+                }
             }
+
+            LifetimeStart = startTime;
```

Comment:
I’m not convinced. Just reading the code you can tell fp.LifetimeStart is going to be different from LifetimeStart since the preempt isn’t included. Have you actually tested this for visual correctness?

I suspect what you want to do is actually use the minimum and maximum lifetimes of the children. All of this stuff is too raw and fragile against these transforms potentially changing (and they will change to match stable at some point).

Reply:
Yeah, this should probably be done in a similar way to `finalTransformEndTime` (based on the first the first `fadeInTime`).

We can't use the old code – it was calling `Expire` from BDL, using the game clock instead of audio clock and resulting in completely warped lifetimes (lasting hundreds of seconds).

Reply:
I jumped the gun here, it does look visually incorrect. Will remember to check better next time.